### PR TITLE
test(number-input): refactor accessibility tests

### PR DESCRIPTION
**Pull request by ZhuoyuJin**
### Proposed behaviour
- Add `accessibility` Cypress tests for the `Number Input` component to use the new cypress-component-react framework for testing.

### Current behaviour
Old approach for accessibility tests

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

- [x] Run `npx cypress open --component` to check if there are newly added tests
- [x] Check if the `number.test.js` file passed
- [x] Run `npx cypress run --component` to check none of the other *.test.js files have regressed
- [x] Run `npx cypress run --e2e` to check none of the feature files have regressed
- [x] Run `npm run build-storybook` -> `npx sb extract` -> `npx http-server storybook-static -p 9001` and run `npx cypress run --e2e` and check that `Number` tests are not running twice.

### CODE REVIEW

File: src/components/number/number.test.js [hunk 1]
```diff
@@ -346,4 +333,169 @@ context("Tests for Number component", () => {
...
+      }
+    );
+
+    it.each(["11", "10"])(
+      "should check accessibility when enforce character limit of %s in Number",
+      (limit) => {
+        CypressMountWithProviders(
+          <NumberInputComponent enforceCharacterLimit characterLimit={limit} />
+        );
+
```
Comment (DlgSHi) [inline src/components/number/number.test.js:397]: we are not checking this properly - we need to type something over limit to check accessibility

File: src/components/number/number.test.js [hunk 2]
```diff
@@ -346,4 +333,169 @@ context("Tests for Number component", () => {
...
+      "should check accessibility and warn if over character limit of %s in Number",
+      (limit) => {
+        CypressMountWithProviders(
+          <NumberInputComponent
+            enforceCharacterLimit={false}
+            warnOverLimit
+            characterLimit={limit}
+          />
+        );
+
```
Comment (DlgSHi) [inline src/components/number/number.test.js:412]: same here

File: src/components/number/number.test.js [hunk 3]
```diff
@@ -346,4 +333,169 @@ context("Tests for Number component", () => {
...
+    });
+
+    it("should check accessibility when Number is read only", () => {
+      CypressMountWithProviders(<NumberInputComponent readOnly />);
+
+      cy.checkAccessibility();
+    });
+
+    it.each([SIZE.SMALL, SIZE.MEDIUM, SIZE.LARGE])(
+      "should use %s as size and render Number with %s as height",
```
Comment (DlgSHi) [inline src/components/number/number.test.js:436]: we don't have a second parameter here so second `%s` will be empty

File: src/components/number/number.test.js [hunk 4]
```diff
@@ -346,4 +333,169 @@ context("Tests for Number component", () => {
       }
     );
   });
+
+  describe("check accessibility for Number component", () => {
```
Comment (DlgSHi) [inline src/components/number/number.test.js:337]: isn't it easier to import just all stories from `number.stories.tsx` file? and just check it?

**Review by DlgSHi — CHANGES_REQUESTED** [review]

**Review by DlgSHi — APPROVED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Pull request merged**